public/lectures_list.php

```diff
@@ -0,0 +1,108 @@
+<?php require_once(dirname(__FILE__)."/../modules/ensure_logged_in.php"); ?>
+
+<?php
+    require "../modules/models/course.php";
+    require "../modules/models/lecture.php";
+
+    $course_id = $_GET['id'];
+
+    try {
+        $specific_course = Course::where(array('id' => $course_id));
+    } catch (PDOException $error) {
+        echo "<br>" . $error->getMessage();
+    }
+
+    try {
+        $existing_lectures = Lecture::where(array('course_id' => $course_id));
+    } catch (PDOException $error) {
+        echo "<br>" . $error->getMessage();
+    }
+
+    if (!empty($_POST)) {
+        if (get_current_role() == "admin") {        
+            
+            if (isset($_POST["submitLecture"])) {
+            $new_lecture = new Lecture();
+            $new_lecture->course_id = $course_id;
+            $new_lecture->lecture_code = $_POST['lecture_code'];
+            $new_lecture->starting_date = $_POST['starting_date'];
+            $new_lecture->ending_date = $_POST['ending_date'];
+
+            try {
+                if ($new_lecture->starting_date > $new_lecture->ending_date) {
+                    throw new PDOException("The ending date cannot be before the beginning.");
+                }
+                $new_lecture->save();
+                $create_success = true;
+            } catch (PDOException $error) {
+                echo "The Lecture could not be added!<br>Error: " . $error->getMessage();
+                }
+            } 
+        }
+    }
+
+    if (isset($_POST['submitLecture']) && isset($create_success) && $create_success) {
+            header('location:lectures_list.php?id='.$course_id);
+        }
+?>
+
+<?php include "templates/header.php"; ?>
+
+<html>
+        <h2>Lectures for <?php echo($specific_course[0]->course_code), " - ", ($specific_course[0]->course_name);?></h2>
```

Comment:
`$specific_course[0]` you're taking it for granted that the `course_id` is always legitimate, the result set could be empty